java-checks/src/main/java/org/sonar/java/checks/ShiftOnIntOrLongCheck.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * dev@sonar.codehaus.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import org.sonar.check.BelongsToProfile;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.AbstractTypedTree;
+import org.sonar.java.resolve.Type;
+import org.sonar.plugins.java.api.tree.ArrayAccessExpressionTree;
+import org.sonar.plugins.java.api.tree.AssignmentExpressionTree;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.plugins.java.api.tree.UnaryExpressionTree;
+
+import java.text.MessageFormat;
+import java.util.List;
+
+@Rule(
+  key = "S2183",
+  priority = Priority.CRITICAL,
+  tags = {"bug"})
+@BelongsToProfile(title = "Sonar way", priority = Priority.CRITICAL)
+public class ShiftOnIntOrLongCheck extends SubscriptionBaseVisitor {
+
+  private static final String USELESS_SHIFT_MESSAGE = "Remove this useless shift (multiple of {0})";
+  private static final String INTEGER_NO_ID_MESSAGE = "Either use a \"long\" or correct this shift to {0}";
+  private static final String INTEGER_ID_MESSAGE = "Either make \"{1}\" a \"long\" or correct this shift to {0}";
+  private static final String LONG_MESSAGE = "Correct this shift to {0}";
```

Comment:
Definitely : messages should be inlined in code. It is not going to be reused and it makes the code difficult to read as it is indirected if you put them as constants.